File: frontend-mobile/services/supabase.ts
import { createClient, SupabaseClient } from '@supabase/supabase-js';
import Constants from 'expo-constants';

// Supabase 클라이언트 초기화 (앱 시작 시 즉시 초기화)
let supabaseClient: SupabaseClient | null = null;

function initializeSupabaseClient(): SupabaseClient {
  if (supabaseClient) {
    return supabaseClient;
  }

  // 환경 변수 읽기 (우선순위 순서)
  // 1. process.env (Expo가 자동으로 .env 파일을 로드)
  // 2. Constants.expoConfig.extra (app.json의 extra 섹션)
  // 3. Constants.manifest?.extra (구버전 호환)
  const SUPABASE_URL = 
    process.env.EXPO_PUBLIC_SUPABASE_URL || 
    Constants.expoConfig?.extra?.EXPO_PUBLIC_SUPABASE_URL ||
    Constants.manifest?.extra?.EXPO_PUBLIC_SUPABASE_URL ||
    (global as any).__EXPO_PUBLIC_SUPABASE_URL;
    
  const SUPABASE_ANON_KEY = 
    process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY || 
    Constants.expoConfig?.extra?.EXPO_PUBLIC_SUPABASE_ANON_KEY ||
    Constants.manifest?.extra?.EXPO_PUBLIC_SUPABASE_ANON_KEY ||
    (global as any).__EXPO_PUBLIC_SUPABASE_ANON_KEY;


  if (!SUPABASE_URL || !SUPABASE_ANON_KEY) {
    // 더미 클라이언트 생성 (에러 방지, 실제 사용 시 에러 발생)
    supabaseClient = createClient('https://placeholder.supabase.co', 'placeholder-key');
    return supabaseClient;
  }

  supabaseClient = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);
  return supabaseClient;
}

// 앱 시작 시 즉시 초기화
initializeSupabaseClient();

// Proxy를 사용하여 초기화된 클라이언트 접근
export const supabase = new Proxy({} as SupabaseClient, {
  get(_target, prop) {
    const client = initializeSupabaseClient();
    const value = client[prop as keyof SupabaseClient];
    if (typeof value === 'function') {
      return value.bind(client);
    }
    return value;
  },
});

File: backend/app/db/supabase.py
from supabase import create_client, Client
from app.core.config import settings

if not settings.SUPABASE_URL or not settings.SUPABASE_ANON_KEY or not settings.SUPABASE_SERVICE_ROLE_KEY:
    raise ValueError(
        f"Supabase 환경 변수가 설정되지 않았습니다.\n"
        f"SUPABASE_URL: {'SET' if settings.SUPABASE_URL else 'NOT SET'}\n"
        f"SUPABASE_ANON_KEY: {'SET' if settings.SUPABASE_ANON_KEY else 'NOT SET'}"
        f"SUPABASE_SERVICE_ROLE_KEY: {'SET' if settings.SUPABASE_SERVICE_ROLE_KEY else 'NOT SET'}"
    )

def get_supabase_client() -> Client:
    return create_client(
        settings.SUPABASE_URL,
        settings.SUPABASE_ANON_KEY
    )

supabase: Client = get_supabase_client()

def get_supabase_admin_client() -> Client:
    return create_client(
        settings.SUPABASE_URL,
        settings.SUPABASE_SERVICE_ROLE_KEY
    )

supabase_admin: Client = get_supabase_admin_client()

print("========== Supabase ENV CHECK ==========")
print("SUPABASE_URL:", settings.SUPABASE_URL)
print("SUPABASE_ANON_KEY:", "SET" if settings.SUPABASE_ANON_KEY else "NOT SET")
print("SUPABASE_SERVICE_ROLE_KEY:", "SET" if settings.SUPABASE_SERVICE_ROLE_KEY else "NOT SET")
print("=======================================")